src/validation.cpp

```diff
@@ -2509,18 +2509,15 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
     uint256 hashPrevBlock = pindex->pprev == NULL ? uint256() : pindex->pprev->GetBlockHash();
     assert(hashPrevBlock == view.GetBestBlock());
 
-    // Gesesis coinbase *is* spendable
+    // Add genesis outputs, if any from last transaction.
     if (block.GetHash() == chainparams.GetConsensus().hashGenesisBlock) {
         if (!fJustCheck) {
-            assert(block.vtx.size() == 1);
-            assert(block.nHeight == 0);
-
             std::vector<std::pair<uint256, CDiskTxPos> > vPos;
             std::multimap<uint256, std::pair<COutPoint, CAmount> > mLocksCreated;
-            const CTransaction tx = *(block.vtx[0]);
 
-            CTxUndo undoDummy;
-            UpdateCoins(tx, view, undoDummy, pindex->nHeight);
+            const CTransaction tx = *(block.vtx[block.vtx.size()-1]);
```

Comment:
Should this be a loop over all coinbase transactions? If we ever have a situation where `AppendPolicyIssuance` is called more than once, this code means that only the last one's outputs are spendable.